Fails while tagging new release if remote was updated during the release

## Current behavior

1. A build with semantic release is triggered. 
2. Semantic release analyzes the commits and determines a version number to tag
3. The remote branch gets updated, while semantic-release is still running (maybe a longer `@semantic-release/exec` command)
3. All commands finish and semantic-release tries to push the new tag to the remote but fails: 
```
Error: Command failed: git push --tags https://remote.gitrepo.com/path/to/project HEAD:master
To https://remote.gitrepo.com/path/to/project
  * [new tag]         v1.1.8 -> v1.1.8
  ! [rejected]        HEAD -> master (fetch first)
 error: failed to push some refs to 'https://remote.gitrepo.com/path/to/project'
 hint: Updates were rejected because the remote contains work that you do
 hint: not have locally. This is usually caused by another repository pushing
 hint: to the same ref. You may want to first integrate the remote changes
 hint: (e.g., 'git pull ...') before pushing again.
 hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

## Expected behavior

Semantic Release pushes the Tag and finishes the build succesfully, even if the remote is updated in the meanwhile. 

## Workaround

Do not push to master while your project is building. 

## Environment

- **semantic-release** version: 15.10.3
- CI environment: Team Foundation Service (Azure DevOps OnPrem)
- Plugins used: `@semantic-release/changelog`, `@semantic-release/exec`, `@iteratec/semantic-release-docker`
- **semantic-release** configuration: 
```
"release": {
    "branch": "master",
    "verifyConditions": [],
    "verifyConfig": [
      "@iteratec/semantic-release-docker"
    ],
    "prepare": [
      {
        "path": "@semantic-release/changelog",
        "changelogFile": "dist/changelog.md"
      },
      {
        "path": "@semantic-release/exec",
        "cmd": "node version.js ${nextRelease.version} ${commits[0].commit.short} ${commits[0].commit.long}"
      },
      {
        "path": "@semantic-release/exec",
        "cmd": "docker build . -f Dockerfile-dist -t dist_project"
      },
      {
        "path": "@semantic-release/exec",
        "cmd": "docker build ./infrastructure/nginx -t project"
      },
      {
        "path": "@iteratec/semantic-release-docker",
        "additionalTags": [
          "latest"
        ],
        "imageName": "project",
        "registryUrl": "remote.registry.com",
        "respositoryName": "project"
      }
    ],
    "publish": [
      {
        "path": "@iteratec/semantic-release-docker"
      }
    ],
    "fail": [],
    "success": []
  }
```
- CI logs: 
```
2018-11-20T14:01:33.8799780Z [14:01:33] [semantic-release] › ✖  An error occurred while running semantic-release: { Error: Command failed: git push --tags https://remote.gitrepo.com/path/to/project HEAD:master
2018-11-20T14:01:33.8823860Z To https://remote.gitrepo.com/path/to/project
2018-11-20T14:01:33.8844900Z  * [new tag]         v1.1.8 -> v1.1.8
2018-11-20T14:01:33.8865570Z  ! [rejected]        HEAD -> master (fetch first)
2018-11-20T14:01:33.8886290Z error: failed to push some refs to 'https://remote.gitrepo.com/path/to/project'
2018-11-20T14:01:33.8905530Z hint: Updates were rejected because the remote contains work that you do
2018-11-20T14:01:33.8926550Z hint: not have locally. This is usually caused by another repository pushing
2018-11-20T14:01:33.8944380Z hint: to the same ref. You may want to first integrate the remote changes
2018-11-20T14:01:33.8964060Z hint: (e.g., 'git pull ...') before pushing again.
2018-11-20T14:01:33.8984530Z hint: See the 'Note about fast-forwards' in 'git push --help' for details.
2018-11-20T14:01:33.8995090Z 
2018-11-20T14:01:33.9003430Z 
2018-11-20T14:01:33.9043480Z     at makeError (/vsts/agent/_work/4/s/node_modules/execa/index.js:174:9)
2018-11-20T14:01:33.9061320Z     at Promise.all.then.arr (/vsts/agent/_work/4/s/node_modules/execa/index.js:278:16)
2018-11-20T14:01:33.9078110Z     at <anonymous>
2018-11-20T14:01:33.9094970Z     at process._tickCallback (internal/process/next_tick.js:188:7)
2018-11-20T14:01:33.9111570Z   code: 1,
2018-11-20T14:01:33.9144140Z   stdout: '',
2018-11-20T14:01:33.9170470Z   stderr: 'To https://remote.gitrepo.com/path/to/project\n * [new tag]         v1.1.8 -> v1.1.8\n ! [rejected]        HEAD -> master (fetch first)\nerror: failed to push some refs to \'https://remote.gitrepo.com/path/to/project\'\nhint: Updates were rejected because the remote contains work that you do\nhint: not have locally. This is usually caused by another repository pushing\nhint: to the same ref. You may want to first integrate the remote changes\nhint: (e.g., \'git pull ...\') before pushing again.\nhint: See the \'Note about fast-forwards\' in \'git push --help\' for details.\n',
2018-11-20T14:01:33.9195980Z   failed: true,
2018-11-20T14:01:33.9212340Z   signal: null,
2018-11-20T14:01:33.9231630Z   cmd: 'git push --tags https://remote.gitrepo.com/path/to/project HEAD:master',
2018-11-20T14:01:33.9251050Z   timedOut: false,
2018-11-20T14:01:33.9268150Z   killed: false }
2018-11-20T14:01:33.9288320Z { Error: Command failed: git push --tags https://remote.gitrepo.com/path/to/project HEAD:master
2018-11-20T14:01:33.9309460Z To https://remote.gitrepo.com/path/to/project
2018-11-20T14:01:33.9329840Z  * [new tag]         v1.1.8 -> v1.1.8
2018-11-20T14:01:33.9350000Z  ! [rejected]        HEAD -> master (fetch first)
2018-11-20T14:01:33.9371550Z error: failed to push some refs to 'https://remote.gitrepo.com/path/to/project'
2018-11-20T14:01:33.9390980Z hint: Updates were rejected because the remote contains work that you do
2018-11-20T14:01:33.9407790Z hint: not have locally. This is usually caused by another repository pushing
2018-11-20T14:01:33.9424640Z hint: to the same ref. You may want to first integrate the remote changes
2018-11-20T14:01:33.9443230Z hint: (e.g., 'git pull ...') before pushing again.
2018-11-20T14:01:33.9463680Z hint: See the 'Note about fast-forwards' in 'git push --help' for details.
2018-11-20T14:01:33.9474210Z 
2018-11-20T14:01:33.9482510Z 
2018-11-20T14:01:33.9499090Z     at makeError (/vsts/agent/_work/4/s/node_modules/execa/index.js:174:9)
2018-11-20T14:01:33.9515870Z     at Promise.all.then.arr (/vsts/agent/_work/4/s/node_modules/execa/index.js:278:16)
2018-11-20T14:01:33.9532630Z     at <anonymous>
2018-11-20T14:01:33.9550010Z     at process._tickCallback (internal/process/next_tick.js:188:7)
2018-11-20T14:01:33.9568250Z   code: [33m1[39m,
2018-11-20T14:01:33.9589140Z   stdout: [32m''[39m,
2018-11-20T14:01:33.9615600Z   stderr: [32m'To https://remote.gitrepo.com/path/to/project\n * [new tag]         v1.1.8 -> v1.1.8\n ! [rejected]        HEAD -> master (fetch first)\nerror: failed to push some refs to \'https://remote.gitrepo.com/path/to/project\'\nhint: Updates were rejected because the remote contains work that you do\nhint: not have locally. This is usually caused by another repository pushing\nhint: to the same ref. You may want to first integrate the remote changes\nhint: (e.g., \'git pull ...\') before pushing again.\nhint: See the \'Note about fast-forwards\' in \'git push --help\' for details.\n'[39m,
2018-11-20T14:01:33.9641080Z   failed: [33mtrue[39m,
2018-11-20T14:01:33.9660980Z   signal: [1mnull[22m,
2018-11-20T14:01:33.9682130Z   cmd: [32m'git push --tags https://remote.gitrepo.com/path/to/project HEAD:master'[39m,
2018-11-20T14:01:33.9702800Z   timedOut: [33mfalse[39m,
2018-11-20T14:01:33.9763440Z   killed: [33mfalse[39m }npm ERR! code ELIFECYCLE
2018-11-20T14:01:33.9781990Z npm ERR! errno 1
2018-11-20T14:01:33.9800530Z npm ERR! project@0.4.0 release: `semantic-release "--no-ci"`
2018-11-20T14:01:33.9818900Z npm ERR! Exit status 1
2018-11-20T14:01:33.9835210Z npm ERR! 
2018-11-20T14:01:33.9852180Z npm ERR! Failed at the project@0.4.0 release script.
2018-11-20T14:01:33.9869170Z npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
2018-11-20T14:01:33.9877910Z 
2018-11-20T14:01:33.9894410Z npm ERR! A complete log of this run can be found in:
2018-11-20T14:01:33.9913050Z npm ERR!     /root/.npm/_logs/2018-11-20T14_01_33_892Z-debug.log
2018-11-20T14:01:34.0111640Z ##[error]npm failed with error: /usr/local/bin/npm failed with return code: 1
```

--- a/lib/git.js
+++ b/lib/git.js
@@ -149,7 +149,7 @@ async function tag(tagName, execaOpts) {
  * @throws {Error} if the push failed.
  */
 async function push(repositoryUrl, branch, execaOpts) {
-  await execa('git', ['push', '--tags', repositoryUrl, `HEAD:${branch}`], execaOpts);
+  await execa('git', ['push', '--tags', repositoryUrl], execaOpts);
 }
 
 /**

--- a/test/git.test.js
+++ b/test/git.test.js
@@ -204,19 +204,18 @@ test('Return "true" if repository is up to date', async t => {
 });
 
 test('Return falsy if repository is not up to date', async t => {
-  let {cwd, repositoryUrl} = await gitRepo(true);
-  const repoDir = cwd;
+  const {cwd, repositoryUrl} = await gitRepo(true);
   await gitCommits(['First'], {cwd});
   await gitCommits(['Second'], {cwd});
   await gitPush(repositoryUrl, 'master', {cwd});
 
   t.true(await isBranchUpToDate('master', {cwd}));
 
-  cwd = await gitShallowClone(repositoryUrl);
-  await gitCommits(['Third'], {cwd});
-  await gitPush('origin', 'master', {cwd});
+  const tmpRepo = await gitShallowClone(repositoryUrl);
+  await gitCommits(['Third'], {cwd: tmpRepo});
+  await gitPush('origin', 'master', {cwd: tmpRepo});
 
-  t.falsy(await isBranchUpToDate('master', {cwd: repoDir}));
+  t.falsy(await isBranchUpToDate('master', {cwd}));
 });
 
 test('Return "true" if local repository is ahead', async t => {

--- a/test/git.test.js
+++ b/test/git.test.js
@@ -151,10 +151,24 @@ test('Add tag on head commit', async t => {
   await t.is(await gitCommitTag(commits[0].hash, {cwd}), 'tag_name');
 });
 
-test('Push tag and commit to remote repository', async t => {
+test('Push tag to remote repository', async t => {
   // Create a git repository with a remote, set the current working directory at the root of the repo
   const {cwd, repositoryUrl} = await gitRepo(true);
   const commits = await gitCommits(['Test commit'], {cwd});
+
+  await tag('tag_name', {cwd});
+  await push(repositoryUrl, 'master', {cwd});
+
+  t.is(await gitRemoteTagHead(repositoryUrl, 'tag_name', {cwd}), commits[0].hash);
+});
+
+test('Push tag to remote repository with remote branch ahaed', async t => {
+  const {cwd, repositoryUrl} = await gitRepo(true);
+  const commits = await gitCommits(['First'], {cwd});
+  await gitPush(repositoryUrl, 'master', {cwd});
+  const tmpRepo = await gitShallowClone(repositoryUrl);
+  await gitCommits(['Second'], {cwd: tmpRepo});
+  await gitPush('origin', 'master', {cwd: tmpRepo});
 
   await tag('tag_name', {cwd});
   await push(repositoryUrl, 'master', {cwd});

